src/App/Repository/Form/StatsRepository.php

```diff
@@ -260,12 +281,62 @@ public function getRecipients($form_id)
         ];
         $query = $this->selectQuery($where)
             ->resetSelect()
-            ->select([DB::expr('COUNT(distinct contact_id)'), 'total']);
+            ->select([DB::expr('COUNT(contacts.id)'), 'total']);
+
         $query = $this->targetedSurveyStateJoin($query);
+        if ($created_after || $created_before) {
+            $query
+                ->join('messages', 'INNER')
+                ->on('messages.contact_id', '=', 'contacts.id')
+                ->where('messages.direction', '=', 'outgoing');
+            if ($created_after) {
+                $query->where('messages.created', '>=', $created_after);
+            }
+            if ($created_before) {
+                $query->where('messages.created', '<=', $created_before);
+            }
+        }
+        
         return $query
             ->execute($this->db)
             ->get('total');
     }
+    /**
+     * @param $query
+     * @param int $form_id
+     * @param string $created_after
+     * @param string $created_before
+     * @param $result
+     * @return array
+     */
+    public function getResponseRecipients($form_id, $created_after, $created_before)
+    {
+        
+        $query = DB::select()
+            ->from('posts')
+            ->where('form_id', '=', $form_id);
+
+        if ($created_after) {
+            $query = DB::select()
+            ->from([$query, 'form_posts'])
+            ->where('created', '>=', $created_after);
+        }
+
+        if ($created_before) {
+            $query = DB::select()
+            ->from([$query, 'selected_posts'])
+            ->where('created', '<=', $created_before);
+        }
+
+        $query = DB::select('contact_id')
```

Comment:
this overrides the $query instances before it . Is it intended to add to it and then execute, instead?

Reply:
I removed the unnecessary ones in the conditionals that check for created_after/before, but we still need the final one. it won't overwrite it because you're passing it in as the first argument in the `from`.